libs/coin-modules/coin-evm/src/__tests__/coin-tester/scenario-1.test.ts

```diff
@@ -0,0 +1,269 @@
+import { getCryptoCurrencyById } from "@ledgerhq/cryptoassets/currencies";
+import { buildAccountBridge, buildCurrencyBridge } from "../../bridge/js";
+import { ethereum, ERC20Interface, USDC_ON_ETHEREUM, ERC721Interface } from "./helpers";
+import { clearExplorerAppendix, getLogs, setBlock } from "./indexer";
+import { makeAccount } from "../fixtures/common.fixtures";
+import { EvmNftTransaction, Transaction as EvmTransaction } from "../../types";
+import resolver from "../../hw-getAddress";
+import { killAnvil, spawnAnvil } from "./docker";
+import { setCoinConfig } from "../../config";
+import { SignOperationEvent } from "@ledgerhq/types-live";
+import axios from "axios";
+
+const nonce = 0;
+
+const scenarioSendTransaction: ScenarioTransaction<EvmTransaction> = {
+  name: "Send ethereum",
+  amount: new BigNumber(100),
+  useAllAmount: false,
+  recipient: "0x6bfD74C0996F269Bcece59191EFf667b3dFD73b9",
+  feesStrategy: "medium",
+  family: "evm",
+  mode: "send",
+  gasPrice: new BigNumber(0),
+  gasLimit: new BigNumber(21000),
+  nonce,
+  chainId: 1,
+};
+
+const scenarioERC721Transaction: ScenarioTransaction<EvmTransaction & EvmNftTransaction> = {
+  name: "Send NFT",
+  amount: new BigNumber(1),
+  useAllAmount: false,
+  recipient: "0x6bfD74C0996F269Bcece59191EFf667b3dFD73b9",
+  feesStrategy: "medium",
+  family: "evm",
+  mode: "erc721",
+  nft: {
+    tokenId: "3368",
+    contract: "0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D",
+    quantity: new BigNumber(1),
+    collectionName: "Bored Ape",
+  },
+  gasPrice: new BigNumber(0),
+  gasLimit: new BigNumber(21000),
+  nonce: nonce + 1,
+  chainId: 1,
+};
+
+const defaultNanoAppVersion = { firmware: "2.2.3" as const, version: "1.10.4" as const };
+const scenarioEthereum: Scenario<EvmTransaction> = {
+  setup: async () => {
+    const [transport] = await Promise.all([
+      spawnSigner(
+        "speculos",
+        `/${defaultNanoAppVersion.firmware}/Ethereum/app_${defaultNanoAppVersion.version}.elf`,
+      ),
+      spawnAnvil("https://rpc.ankr.com/eth"),
+    ]);
+
+    const provider = new providers.StaticJsonRpcProvider("http://127.0.0.1:8545");
+    const signerContext = (deviceId: string, fn: any): any => fn(new Eth(transport));
+    const onSignerConfirmation = async (e?: SignOperationEvent): Promise<void> => {
+      if (e?.type === "device-signature-requested") {
+        const { data } = await axios.get(
+          `http://localhost:${process.env.API_PORT}/events?currentscreenonly=true`,
+        );
+
+        if (data.events[0].text !== "Accept") {
+          await axios.post(`http://localhost:${process.env.API_PORT}/button/right`, {
+            action: "press-and-release",
+          });
+          onSignerConfirmation(e);
+        } else {
+          await axios.post(`http://localhost:${process.env.API_PORT}/button/both`, {
+            action: "press-and-release",
+          });
+        }
+      }
+    };
```

Comment:
Still think we should extract that from the setup, it feels messy here I think 🤔